src/mscorlib/shared/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -357,6 +357,20 @@ public static void Write<T>(ref byte destination, T value)
             return ref Unsafe.As<byte, T>(ref *(byte*)source);
         }
 
+        /// <summary>
+        /// Reinterprets the given location as a reference to a value of type <typeparamref name="T"/>.
+        /// </summary>
+        [Intrinsic]
+        [NonVersionable]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ref T AsRef<T>(in T source)
```

Comment:
This works by a pure luck. The method may or may not get the overridden IL on the VM side, depending on how it is ordered with other AsRef.

You would need to disambiguate this with the other AsRef in the VM, and supply the IL for this.

Reply:
Somehow it works, but I have to agree here - if there is a way to do it correctly, it should be done correctly.